### PR TITLE
fix some misc hostdb errors

### PR DESCRIPTION
**Pull request by DavidVorick**
Should fix the 'negative currency' reports - problem was due to a race
condition.

Tree is sorted to have better structure.

Some style adjustments

Some consistency code was added during debugging. While useful, right
now it's not actually used anywhere, so it's commented out. Will
probably come in handy later, so I left it in this commit.

### CODE REVIEW

File: modules/renter/hostdb/scan.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func (hdb *HostDB) threadedProbeHosts() {
 
 			// If 'maxActiveHosts' has not been reached, add the host to the
 			// activeHosts tree.
-			if _, exists := hdb.activeHosts[hostEntry.NetAddress]; !exists && len(hdb.activeHosts) < maxActiveHosts {
+			if _, exists := hdb.activeHosts[hostEntry.NetAddress]; exists || len(hdb.activeHosts) < maxActiveHosts {
```
Comment (DavidVorick) [inline modules/renter/hostdb/scan.go]: This guy was causing the negative currency errors and host tree restructuring. Lack of safety on the hostdb tree can be blamed for this.

Adding safety would be a pretty major refactor.

File: modules/renter/hostdb/persist.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func (hdb *HostDB) load() error {
 		hdb.allHosts[data.AllHosts[i].NetAddress] = &data.AllHosts[i]
 	}
 	for i := range data.ActiveHosts {
-		hdb.insertNode(&data.AllHosts[i])
+		hdb.insertNode(&data.ActiveHosts[i])
```
Comment (lukechampine) [inline modules/renter/hostdb/persist.go]: :sob:

File: modules/renter/hostdb/weightedlist_consistency_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		fmt.Println("No Left")
+	}
+	if hn.right != nil {
+		fmt.Println("Node Right")
+		hn.right.Print()
+	} else {
+		fmt.Println("No Right")
+	}
+}
+*/
```
Comment (lukechampine) [inline modules/renter/hostdb/weightedlist_consistency_test.go]: what's the deal with these? If they're actually helpful, they shouldn't be commented out.

Comment (DavidVorick) [inline modules/renter/hostdb/weightedlist_consistency_test.go]: They are helpful for debugging, but not necessarily for testing. For example, printing a node is something you'd never do during testing.

File: modules/renter/hostdb/persist.go (hunk 2 above)
Comment (DavidVorick) [inline modules/renter/hostdb/persist.go]: yeah totally my fault too

File: modules/renter/hostdb/weightedlist_consistency_test.go (hunk 3 above)
Comment (lukechampine) [inline modules/renter/hostdb/weightedlist_consistency_test.go]: just seems weird to me to purposely include commented-out code.

Comment (DavidVorick) [inline modules/renter/hostdb/weightedlist_consistency_test.go]: I could change it to `String`. I can also uncomment them if you prefer.

**Pull request merged**